wasi/wasi.go

```diff
@@ -386,3 +400,9 @@ func environ_sizes_get(*wasm.HostFunctionCallContext, uint32, uint32) (err Errno
 func environ_get(*wasm.HostFunctionCallContext, uint32, uint32) (err Errno) {
 	return
 }
+
+func clock_time_get(ctx *wasm.HostFunctionCallContext, id uint32, precision uint64, timestampPtr uint32) (err Errno) {
+	nanos := uint64(time.Now().UnixNano())
+	binary.LittleEndian.PutUint64(ctx.Memory.Buffer[timestampPtr:], nanos)
```

Comment:
Could you define new `ctx.Memory.PutUint64` which checks the given pointer's range and use it instead of `binary.LittleEndian.PutUint64`, please? The low-level `binary.LittleEndian.PutUint64` allows untrusted wasm binary crash the wazero and the host environment by passing an invalid `timestampPtr`.

You can follow the implementation of `ctx.Memory.PutUint32` here: https://github.com/tetratelabs/wazero/blob/5a540c44e6de47d4986396d57c2a1b5cdbb0ba2c/wasm/store.go#L808

Reply:
I know there's some code calling `binary.LittleEndian.PutXXX` around fs APIs, but they will be updated as well soon. Sorry it it confused.

Reply:
Ah cool. Thanks for pointing this out. This makes it so people don’t have to know that Wasm is little endian.

Reply:
Done